crates/bevy_ecs/src/component.rs

```diff
@@ -73,6 +73,14 @@ use derive_more::derive::{Display, Error};
 ///
 /// # Component and data access
 ///
+/// Components can be marked as immutable by adding the `#[component(immutable)]` attribute when using the
+/// derive macro.
+///
+/// Immutable components can be removed, replaced, and inserted just like mutable components.
+/// The only guarantee that's enforced is that while an immutable component is attached
```

Comment:
The framing here is backwards. Talk about the way it's different first, then how it's the same.